content/topics/Analyze/causal-inference/did/goodmanbacon.md

```diff
@@ -0,0 +1,152 @@
+
+The implication of these forbidden comparisons is that our supposedly causal TWFE estimates are partly composed of non-causal, biased coefficients, which actually compare treatment to treatment, as opposed to treatment to control.  
+
+2. Negative weights: each comparison is assigned a weight which is proportional to group size (the number of observations within the group) and variation in treatment exposure. The larger the group size and the smaller the variation in treatment exposure, the higher the attached weight. All weights sum to one, but they can also be negative – sometimes, the comparisons at the tails of the time period distribution (in very early or very late periods) can carry negative weights, as treatment variation is typically smaller there (as most groups are not yet treated in early stages or already treated in late stages).  
+
+{{<katex>}}
+{{</katex>}}
+
+The problem with negative weights is that they can bias our final estimate by altering the direction (sign) of the effect. Let’s suppose we have three comparisons with coefficients of 0.5, 1, and 5 and weights of 0.8, 0.8, and –0.6, respectively. Our total effect is then: $0.5*0.8 + 1*0.8 + 5*(-0.6) = -1.8$. We therefore obtain a negative treatment effect estimate even though each of the individual comparisons yields a positive effect of the treatment! 
+
+## The Goodman-Bacon decomposition 
+
+The Goodman-Bacon decomposition is a diagnostic tool which allows us to see whether our TWFE estimator is reliant on forbidden comparisons or negative weights. The decomposition breaks down the estimator into three types of comparisons: 
+
+1. Treated vs. never treated: this is equivalent to the classic 2x2 DiD comparison. We are only comparing treated to untreated units. 
+
+2. Early treated vs. late control: we are comparing units that are treated early to units that will be treated later, but have not been treated yet. 
+
+3. Late treated vs. early control: we are comparing units that are treated later to units that have _already been treated_ – this is a forbidden comparison! 
+
+{{% tip %}} 
+
+‘Early’ refers to the group that is treated earlier, ‘late’ refers to the group treated later. Therefore, a comparison called ‘early treated’ vs. ‘late control’ simply means that the comparison uses the early group as the treated group and the late group as the control group. In our example, an early treated vs. late control comparison would be city A vs. city B, with city A as the treatment group and city B as the control group. 
+
+{{% /tip %}} 
+
+The decomposition also allows us to see the weight attached to each type of comparison. This way, we are able to see how our overall TWFE estimate is constructed and whether it is prone to the limitations and biases discussed above. In practice, computing the Goodman-Bacon decomposition is very straightforward with the `bacondecomp` [package](https://github.com/evanjflack/bacondecomp) in R.  
+
+## An example in R 

```

Comment:
We discussed earlier about keeping the example consistent throughout the article so that people can connect the code to the intuition but if you feel like the divorce dataset is not straightforward to explain the intuition then lets keep it as is. Either is fine by me :)